amplipi/models.py

```diff
@@ -110,6 +116,114 @@ class fields_w_default(SimpleNamespace):
   Disabled = Field(default=False, description='Set to true if not connected to a speaker')
 
 
+def calc_average(running_average, weight, new_entry):
+  """
+    Calculates an average using three variables:
+
+    running_average, the average so far
+
+    weight, the count of numbers that went into that average
+
+    new_entry, the newest addition to the dataset
+
+
+    Returns the new average, and new weight
+  """
+  return ((running_average * weight) + new_entry) / (weight + 1), weight + 1
+
+
+class StreamUsageSchema(BaseModel):
+  """Schema for individual stream type usage statistics"""
+  # For the purpose of the below comments, a "survey cycle" is defined as how often the device phones home
+
+  # The highest number of concurrent streams running within the survey cycle, constrained to be a number between 0 and 4
+  concurrent_polls: int = 0  # Used to calculate average
+  average_concurrent_streams: Annotated[int, Field(ge=0, le=4)] = 0
+  peak_concurrent_streams: Annotated[int, Field(ge=0, le=4)] = 0
+
+  # The length in seconds of longest-connected instance within the survey cycle, maxing out after 24 hours. Is not additive with multiple concurrent streams.
+  runtime_polls: int = 0  # Used to calculate average
+  average_runtime: Annotated[int, Field(ge=0, le=86400)] = 0
+  peak_runtime: Annotated[int, Field(ge=0, le=86400)] = 0
```

Comment:
For runtime, I've got two ideas for measuring:

1. Measure it by recording the time when a stream starts and recording again when it stops and splitting the difference
2. Have a regularly repeated polling function that just records when a given stream and gives a rough number divisible by the 5-10 minute intervals that that polling function is ran, as specificity isn't important and this doesn't require as many stored variables but does require more than 2 polling instances per stream